dbus/src/nonblock.rs

```diff
@@ -178,13 +178,21 @@ impl Process for $c {
                 return;
             }
         }
-        let ff = self.filters_mut().remove_matching(&msg);
-        if let Some(mut ff) = ff {
-            if ff.2(msg, self) {
+        let matching_filters = self.filters_mut().remove_matching(&msg);
+        if matching_filters.is_empty() {
+            if let Some(reply) = crate::channel::default_reply(&msg) {
```

Comment:
I've kept this to preserve existing behaviour, but I don't understand what it's doing.